metal3-dev/local-cbo.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash -xe
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+
+LOGDIR=${SCRIPTDIR}/logs
+source $SCRIPTDIR/logging.sh
+
+source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/utils.sh
+
+if ! which kubebuilder 2>&1 >/dev/null ; then
+    echo "Did not find kubebuilder." 1>&2
+    exit 1
+fi
+
+if ! which yq 2>&1 >/dev/null ; then
+    echo "Did not find yq" 1>&2
+    echo "Install with: pip3 install --user yq" 1>&2
+    exit 1
+fi
+
+cbo_path=$GOPATH/src/github.com/openshift/cluster-baremetal-operator
+if [ ! -d $cbo_path ]; then
+    echo "Did not find $cbo_path" 1>&2
+    exit 1
+fi
+
+# Stop the machine-api-operator so it does not try to fix the
+# deployment we are going to change.
+$SCRIPTDIR/stop-mao.sh
+
+# Scale the existing deployment down.
+oc scale deployment -n openshift-machine-api --replicas=0 metal3
+if oc get pod -o name -n openshift-machine-api | grep -q metal3; then
+    metal3pods=$(oc get pod -o name -n openshift-machine-api | grep -v metal3-development | grep metal3)
+    oc wait --for=delete -n openshift-machine-api $metal3pods || true
+fi
+
+OUTDIR=${OCP_DIR}/metal3-cbo-dev
+mkdir -p $OUTDIR
+
+# Save a copy of the full deployment as input
+oc get deployment -n openshift-machine-api -o yaml metal3 > $OUTDIR/bmo-deployment-full.yaml
+
+# Add the annotation for the cbo ownership
```

Comment:
The CBO should do this when it comes up. Since the script stops the MAO before starting the CBO there shouldn't be any race condition.